cli/analytics.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  getConfigSettings,
+  getConfigSettingsPath,
+  upsertConfigSettings
+} from "@dataform/cli/config";
+import { question } from "@dataform/cli/console";
+import Analytics = require("analytics-node");
+import { readFile, stat } from "fs";
+import { promisify } from "util";
+import { v4 as uuidv4 } from "uuid";
+
+const analytics = new Analytics("eR24ln3MniE3TKZXkvAkOGkiSN02xXqw");
+
+export async function maybeConfigureAnalytics() {
+  const settings = await getConfigSettings();
+  const dockerStatus = await getDockerStatus();
+  // We should only ask if users want to track analytics if they aren't in docker, e.g. for CI.
+  if (!dockerStatus && settings.allowAnonymousAnalytics === undefined) {
+    const optInResponse = question(
```

Comment:
can you expose a new method in `console.ts` that uses https://www.npmjs.com/package/readline-sync#keyinyn?

```
function ynQuestion(questionText: string, default: boolean = false): boolean {
  const response = readlineSync.keyInYN(questionText);
  if (response === '') {
    return default;
  }
  return response;
}
```